### ISSUE TITLE
OpenIllumination data split

### ISSUE THREAD

**Issue by chensjtu** [eliphatfs/zerorf#8]
Dear authors, thank you for your wonderful work! In this repo, you only provided the scripts on nerf synthetic. I wonder if you can provide the OpenIllumination dataset with train/test split?

**Comment by SarahWeiii** [comment]
We are using `10, 3, 19, 22, 17, 35` for 6-view experiments and `10, 33, 35, 6` for 4-view experiments as training views. The testing split is the same as OpenIllumination test set. All of the scenes share the same cameras.
These idxs are also chosen by kmeans.

**Comment by chensjtu** [comment]
Thanks for your quick reply! This information is very important for our experiments.

**Comment by chensjtu** [comment]
Another concern is that in the paper, the scene used in openillumination is:
<img width="1005" alt="image" src="https://github.com/eliphatfs/zerorf/assets/38242929/015d8504-67db-4b67-93c8-8afb01fbdbf3">
while in the openillumination dataset, there aren't any scene named stone or cake. Thus I wonder the real name of each scene may differ from the link? https://[oppo-us-research.github.io/OpenIllumination/data_olat.json](https://oppo-us-research.github.io/OpenIllumination/data_olat.json) https://[oppo-us-research.github.io/OpenIllumination/data.json](https://oppo-us-research.github.io/OpenIllumination/data.json)

**Comment by SarahWeiii** [comment]
We are using the lighting pattern data (the 013 lighting condition) instead of OLAT data. The object ids are `04, 27, 29, 39, 40, 44, 50, 62`.

**Comment by chensjtu** [comment]
I have tried the KNN algo in the zerorf.py and use the ``stone'' to verify. but the results are not same as you provide.
following code is used to generate the idx:
```ids, poses = get_camera_poses_from_json("lighting_patterns/04_stone/transforms_train.json")
    # in this way, simplily use knn for index select.
    idsnew = kmeans_downsample(poses[..., :3, 3], 4)
    print(idsnew) # [25, 7, 17, 11]
    idsnew = kmeans_downsample(poses[..., :3, 3], 6)
    print(idsnew) # [34, 24, 8, 0, 19, 25]
    print(ids[idsnew[0]], ids[idsnew[1]])
```

The get_camera_poses_from_json is:
```
def get_camera_poses_from_json(json_file):
    imgids, poses = [], []
    meta = load_json(json_file)
    for k, v in meta.items():
        imgid = v['file_path'].split('/')[-1]
        assert imgid == k
        imgids.append(k)
        poses.append(v['transform_matrix'])

    # poses = transform_poses_pca(np.stack(poses))
    return imgids, poses
```
I found that either I use the original transforms_train.json or transform_poses_pca cannot provide the same results. Since we are trying to reproduce the same results as zerorf and compare with zerorf, the accurate setting is vital for us. looking forward to your quick and kind reply!

**Comment by SarahWeiii** [comment]
Kmeans sampling does not guarantee to provide the same results every run. And we transform the camera poses to make their upper direction align with z axis. Are you using our provided ids to run the experiments?
We also provide the openillumination dataloader in the code, named `oppo.py` for reference.

**Comment by chensjtu** [comment]
I see... , But in oppo.py, I cannot access the transforms_aligned.json. To get similar results in zerorf paper, I should use the idx 10, 3, 19, 22, 17, 35 for 6-view experiments and 10, 33, 35, 6 for 4-view (these idx are idx of the transfrom_train.json, starts from 0), and  use func transform_poses_pca to get aligned cameras? btw, oppo dataset is opencv format right?

**Comment by SarahWeiii** [comment]
Yes you should use our ids (for training views). We have uploaded the aligned cameras, you can check them here:
[train views](https://huggingface.co/datasets/OpenIllumination/OpenIllumination/raw/main/others/transforms_alignz_train.json)
[test views](https://huggingface.co/datasets/OpenIllumination/OpenIllumination/raw/main/others/transforms_alignz_train.json)
In this way you should be able to directly use the `oppo.py` as dataloader.

And yes open illumination dataset is opencv format.

**Comment by chensjtu** [comment]
Many thanks for your kind reply! Wish you have a happy new year!
